### DISCUSSION TITLE
Is it possible to call `throw render()` on client side?

### DISCUSSION THREAD

**Discussion by orangecoloured** [vikejs/vike#1385] [Discussion]
I have a component that needs to fetch an image on client side before proceeding further.
Is it possible to do `throw render()` in case of an error in this component or should I move the fetcher and `throw` code to `onBeforeRender`?

Atm I see `Error: AbortRender` in the browser's console, but the error page is not being rendered.

**Comment by brillout** [comment]
What UI framework are you using? The question is whether your UI framework transparently propagates the error so that Vike gets a chance to catch it. For example, I presume React to not do that, although I never tried.

**Reply by orangecoloured** [reply]
I'm using preact.

**Reply by orangecoloured** [reply]
I found a workaround sending a `HEAD` request in `onBeforeRender` to at least check if the requested image url exists and can be loaded.

**Reply by brillout** [reply]
👍 Thanks for circling back on this. Btw. would your company be up for [sponsoring](https://github.com/vikejs/vike/issues/1350)?

**Reply by orangecoloured** [reply]
The company I'm working with does not use vike and and I'm here, because I just decided to try out preact for my personal website. But in case I manage to make proper transition to the new setup, I'll consider sponsoring.
